Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -677,11 +680,8 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         self.snoozeTimingStore.reset()
 
         do {
-            try load(options: startupOptions)
-
-            if (try? tokenStore.fetchToken()) == nil {
-                throw TunnelError.startingTunnelWithoutAuthToken
-            }
+            try await load(options: startupOptions)
+            Logger.networkProtection.log("Startup options loaded correctly")
```

Comment:
The error handling needs to be improved here.  Previously we could only have `startingTunnelWithoutAuthToken`, but now we can get a number of different possible errors from `tokenProvider.adopt` and `tokenProvider.getTokenContainer`.

These errors should ideally implement `CustomNSError` and include underlying error information.  See `PacketTunnelProvider.TunnelError` for reference.

The reason this is important is because it's what'll allow us to debug tunnel start issues coming from these changes, and understand what's going on.

Reply:
Understood, atm getTokenContainer can fail for a variety of reasons, from storage to API, decoding and decryption errors, we don't have a single set Error so implementing this is quite difficult, should I wrap any possible error into a higher level TunnelError?